### ISSUE TITLE
MultiSelect: MultiSelect component has an accessibility issue (Tab key navigation)

### ISSUE THREAD

**Issue by shu-pf** [primefaces/primereact#7790]
### Describe the bug

The MultiSelect component has an accessibility issue.

https://primereact.org/multiselect/

When selecting a value, focus cannot be moved to the next element using the Tab key.
![Image](https://github.com/user-attachments/assets/20ff817b-49c2-4283-b25f-acfa3fbe7059)
![Image](https://github.com/user-attachments/assets/6027a81e-0a27-48f9-a029-982622adcbc1)

This issue occurs because the event handler of the removeTokenIcon in the MultiSelect component stops event propagation for all keydown events.

I understand that preventing event propagation is necessary for certain keys, such as Enter, to avoid unintended form submissions.
However, the current implementation blocks all keydown events, including the Tab key, which prevents users from navigating using the keyboard.

Suggested Fix:

The event handler should only stop propagation for the following keys:
- Space
- Enter
- NumpadEnter

I am happy to contribute a pull request (PR) if needed.

Thank you for maintaining this very useful library!

### System Information

```Shell
System:
    OS: macOS 14.6.1
    CPU: (12) arm64 Apple M2 Pro
    Memory: 1.37 GB / 32.00 GB
    Shell: 3.6.4 - /opt/homebrew/bin/fish
  Binaries:
    Node: 22.13.1 - ~/.nodenv/versions/22.13.1/bin/node
    npm: 10.9.2 - ~/.nodenv/versions/22.13.1/bin/npm
    pnpm: 9.15.5 - ~/.nodenv/versions/22.13.1/bin/pnpm
    Watchman: 2024.12.02.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 134.0.6998.89
    Edge: 134.0.3124.68
    Safari: 17.6
  npmPackages:
    primereact: 10.9.2 => 10.9.2 
    react: 18.2.0 => 18.2.0
```

### Steps to reproduce the behavior

1. Select a value.
2. Move the focus by tab key.
3. Stack focus in remove icon.

### Expected behavior

1. Select a value.
2. Move the focus by tab key.
3. Do not stack focus in remove icon.

**Comment by shu-pf** [comment]
These codes may be reletted for this issue.
https://github.com/primefaces/primereact/blob/d49a0614fe238ebb3fe9d1bfb268e12b61ea751d/components/lib/multiselect/MultiSelect.js#L966
https://github.com/primefaces/primereact/blob/d49a0614fe238ebb3fe9d1bfb268e12b61ea751d/components/lib/multiselect/MultiSelect.js#L781
